graphbolt/src/csc_sampling_graph.cc

```diff
@@ -256,8 +257,75 @@ inline torch::Tensor UniformPick(
       picked_neighbors =
           torch::randint(offset, offset + num_neighbors, {fanout}, options);
     } else {
-      picked_neighbors = torch::randperm(num_neighbors, options);
-      picked_neighbors = picked_neighbors.slice(0, 0, fanout) + offset;
+      picked_neighbors = torch::empty({fanout}, options);
+      AT_DISPATCH_INTEGRAL_TYPES(
+          picked_neighbors.scalar_type(), "UniformPick", ([&] {
+            scalar_t* picked_neighbors_data =
+                picked_neighbors.data_ptr<scalar_t>();
+            // We use different sampling strategies for different sampling case.
+            if (fanout >= num_neighbors / 10) {
+              // Use this algorithm when `fanout >= num_neighbors / 10` to
+              // reduce computation.
+              //
+              // The reservior algorithm is memory-efficient (O(fanout)) but
```

Comment:
Because `Reservoir algorithm` is one of the most classical sampling algorithms, it is usually widely used in data stream sampling. It has better **space complexity**.

I think the main point here is to show how we trade-off the `Reservoir algorithm` and `Fisher-Yates shuffle (current)` algorithms.

Reply:
I see, let's clarify it in the comment.

Reply:
OK, I'll add some necessary comment.